### PR TITLE
Reworked AI to pathfinding link:

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request
-AI now passes along more information about safe target locations and is using better fitting pathfinding tasks when going to buildings All pathfinding calls now return true on reaching their target, and false while walking unlike being mixed before All logic for creating pathing tasks is now consolidated in a dedicated util: EntityNavigationUtils, slimming down the Navigator class a bit.

## Testing
- [ x] Yes I tested this before submitting it.
- [ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/core/entity/ai/workers/guard/DruidCombatAI.java [hunk 1]
```diff
@@ -114,7 +117,7 @@ protected void doAttack(final LivingEntity target)
             if (user.getRandom().nextInt(FLEE_CHANCE) == 0 &&
                   !((AbstractBuildingGuards) user.getCitizenData().getWorkBuilding()).getTask().equals(GuardTaskSetting.GUARD))
             {
-                user.getNavigation().moveAwayFromLivingEntity(target, getAttackDistance() / 2.0, getCombatMovementSpeed());
+                EntityNavigationUtils.walkAwayFrom(user, target.blockPosition(), (int) (getAttackDistance() / 2.0), getCombatMovementSpeed());
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/entity/ai/workers/guard/DruidCombatAI.java:120]: Imports seem weird here/

File: src/main/java/com/minecolonies/core/entity/ai/workers/guard/EntityAIDruid.java [hunk 2]
```diff
@@ -67,16 +68,17 @@ public void guardMovement()
...
+            walkToSafePos(buildingGuards.getGuardPos());
             return;
         }
 
-        if (worker.isWorkerAtSiteWithMove(buildingGuards.getGuardPos(), 10) || Math.abs(buildingGuards.getGuardPos().getY() - worker.blockPosition().getY()) > 3)
+        if ((BlockPosUtil.dist(buildingGuards.getGuardPos(), worker.blockPosition()) <= 10 || walkToSafePos(buildingGuards.getGuardPos()))
+            || Math.abs(buildingGuards.getGuardPos().getY() - worker.blockPosition().getY()) > 3)
         {
-            // Moves the druid randomly to close edges, for better vision to mobs
+            // Moves the ranger randomly to close edges, for better vision to mobs
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/entity/ai/workers/guard/EntityAIDruid.java]: druid was right, no?

File: src/main/java/com/minecolonies/core/entity/ai/workers/production/EntityAIQuarrier.java [hunk 3]
```diff
@@ -669,35 +670,29 @@ public boolean walkToConstructionSite(final BlockPos currentBlock)
     {
         if (workFrom == null)
         {
-            workFrom = findRandomPositionToWalkTo(5, currentBlock);
-            if (workFrom == null && pathBackupFactor > 10)
+            if (EntityNavigationUtils.walkToRandomPosWithin(worker, 5, 1.0, job.findQuarry().getCorners()))
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/entity/ai/workers/production/EntityAIQuarrier.java]: This was supposed to get close to currentBlock and not take a fully random pos, no?

File: src/main/java/com/minecolonies/core/entity/mobs/aitasks/RaiderWalkAI.java [hunk 4]
```diff
@@ -87,43 +87,23 @@ private boolean walk()
...
+                EntityNavigationUtils.walkToPos(raider, moveToPos, 4, false, !moveToPos.equals(targetBlock) && moveToPos.distManhattan(wayPoints.get(0)) > 50 ? 1.8 : 1.1);
+                walkInBuilding = null;
                 randomPathResult = null;
             }
-            else if (walkInBuildingState)
+            else if (walkInBuilding != null)
             {
-                final BlockPos moveToPos = findRandomPositionToWalkTo();
-                if (moveToPos != null)
+                if (EntityNavigationUtils.walkToRandomPosWithin(raider, 10, 0.7, walkInBuilding.getCorners())
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/entity/mobs/aitasks/RaiderWalkAI.java:88]: Does this take the
randomPathResult.getJob().getPathingOptions().withCanEnterDoors(true).withToggleCost(0).withNonLadderClimbableCost(0);

Special costs in account?

File: src/main/java/com/minecolonies/core/entity/pathfinding/pathjobs/PathJobMoveAwayFromLocation.java [hunk 5]
```diff
@@ -135,4 +137,14 @@ public BlockPos getDestination()
     {
         return preferredDirection;
     }
+
+    public static boolean isJobFor(final AbstractPathJob job, final int avoidDistance, final BlockPos toAvoid)
+    {
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/entity/pathfinding/pathjobs/PathJobMoveAwayFromLocation.java:145]: As I don't understand the purpose of this, I think this needs doc =D

**Review by Raycoms — COMMENTED** [review]
Some small nitpicks only

File: src/main/java/com/minecolonies/core/entity/ai/workers/guard/DruidCombatAI.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/entity/ai/workers/guard/DruidCombatAI.java:120]: those are automatic^^

File: src/main/java/com/minecolonies/core/entity/ai/workers/production/EntityAIQuarrier.java (hunk 3 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/entity/ai/workers/production/EntityAIQuarrier.java]: ye should be around current

**Review by Raycoms — DISMISSED** [review]

**Pull request merged**